src/Reflectivity/Breakpoint.class.st

```diff
@@ -17,15 +17,16 @@ Class {
 	#name : #Breakpoint,
 	#superclass : #Object,
 	#instVars : [
-		'condition',
-		'link',
-		'node',
-		'level',
-		'options'
+		'#condition',
+		'#link',
+		'#node',
+		'#level',
+		'#options',
+		'#targetInstance => WeakSlot'
```

Comment:
Do you have a test which explains why it should be weak?
My first thought is that it is wrong. There is a registry of breakpoints in the system. And I would expect that these registry is a weak collection. So breakpoints and their targets will be collected as a garbage when nobody references target objects (considering that target objects always reference breakpoints by design)

Reply:
Target objects are not aware of their possible instrumentation by a breakpoint.
By design, tools only are aware, and it is perfectly ok to prevent garbage collection if the tools requires it.

Breakpoint are only here to break execution of the target object and they should not interfere with garbage collection, hence the weak slot.

I think it's fine to have ghost breakpoints that do nothing because they lost their target.
We could simply check if such breakpoint exist when adding a new breakpoint and clean the registry.

Reply:
Ok. But test is really important for such logic.

Reply:
There are 2 tests, look at first commit.
It's pretty basic, but what else should I test?